implementations/rs/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+use crate::wrap::wrap_info::get_manifest;
+use connection::WalletError;
+use connections::Connections;
+use ethers::{
+    prelude::SignerMiddleware,
+    providers::{Http, Middleware, PendingTransaction, Provider},
+    signers::Signer,
+    types::{
+        transaction::{eip2718::TypedTransaction, eip712::TypedData},
+        TransactionRequest, TxHash,
+    },
+};
+use handler::{
+    handle_call_params, handle_get_block_by_number_params, handle_send_params,
+    handle_sign_typed_data_args,
+};
+use polywrap_core::invoker::Invoker;
+use polywrap_plugin::{
+    error::PluginError,
+    implementor::plugin_impl,
+    JSON::{self, from_str, from_value, to_string, to_value, Value},
+};
+use std::{str::FromStr, sync::Arc};
+use tokio::runtime::Runtime;
+use wrap::module::{
+    ArgsRequest, ArgsSignMessage, ArgsSignTransaction, ArgsSignerAddress, ArgsWaitForTransaction,
+    Module,
+};
+
+pub mod connection;
+pub mod connections;
+mod handler;
+mod networks;
+mod wrap;
+
+#[derive(Debug)]
+pub struct EthereumWalletPlugin {
+    connections: Connections,
+}
+
+impl EthereumWalletPlugin {
+    pub fn new(connections: Connections) -> Self {
+        Self { connections }
+    }
+
+    fn parse_parameters(&self, method: &str, params: &Option<String>) -> Vec<Value> {
+        if self.is_transaction_method(method) {
+            if let Some(parameters) = params {
+                match method {
+                    "eth_call" => handle_call_params(parameters),
+                    _ => handle_send_params(parameters),
+                }
+            } else {
+                panic!("Method {method} needs params")
+            }
+        } else if let Some(parameters) = params {
+            match method {
+                "eth_getBlockByNumber" => handle_get_block_by_number_params(parameters),
+                "eth_feeHistory" => vec![],
+                "eth_signTypedData_v4" => handle_sign_typed_data_args(parameters),
+                _ => from_str(parameters).unwrap(),
+            }
+        } else {
+            vec![]
+        }
+    }
+
+    fn is_transaction_method(&self, method: &str) -> bool {
+        let transaction_methods = ["eth_sendTransaction", "eth_estimateGas", "eth_call"];
+        transaction_methods.contains(&method)
+    }
+}
+
+#[plugin_impl]
+impl Module for EthereumWalletPlugin {
+    fn request(&mut self, args: &ArgsRequest, _: Arc<dyn Invoker>) -> Result<String, PluginError> {
+        let connection = self.connections.get_connection(args.connection.clone());
```

Comment:
Is the clone necessary here, or can a const ref be used?

Reply:
when we remove the clone we get this error:
![image](https://github.com/polywrap/ethereum-wallet/assets/21031138/a727895b-b96b-4473-9218-cf2f67f861db)
so I just added the clone because couldn't find a better approach - open to suggestions